react-exercise/src/pages/userProfile/UserProfile.tsx

```diff
@@ -0,0 +1,66 @@
+import { Link, useParams } from "react-router-dom";
+import Topper from "../../components/topper/Topper";
+import { User } from "../../types/user.ts";
+import defaultProfilePicture from "../../assets/default-profile-picture.svg";
+import Tabs from "../../components/tabs/Tabs.tsx";
+import backArrowIcon from "../../assets/back-arrow-icon.svg";
+import "./UserProfile.scss";
+
+interface UserProfileProps {
+  users: User[];
+}
+
+function UserProfile({ users }: UserProfileProps) {
+  const { id } = useParams();
+  const user = users.find((user) => user.id.toString() === id);
+
+  if (!user) {
+    return <p>User not found yet...</p>;
```

Comment:
This approach to use the loading of the data is fine?

Reply:
It's more of error handling then loading in this case I think, but let's talk F2F